examples/pie_and_polar_charts/bar_of_pie.py

```diff
@@ -19,40 +19,36 @@
 fig.subplots_adjust(wspace=0)
 
 # pie chart parameters
-ratios = [.27, .56, .17]
+overall_ratios = [.27, .56, .17]
 labels = ['Approve', 'Disapprove', 'Undecided']
 explode = [0.1, 0, 0]
 # rotate so that first wedge is split by the x-axis
-angle = -180 * ratios[0]
-ax1.pie(ratios, autopct='%1.1f%%', startangle=angle,
-        labels=labels, explode=explode)
+angle = -180 * overall_ratios[0]
+wedges, *_ = ax1.pie(overall_ratios, autopct='%1.1f%%', startangle=angle,
+                     labels=labels, explode=explode)
 
 # bar chart parameters
-
-xpos = 0
-bottom = 0
-ratios = [.33, .54, .07, .06]
+approved_ratios = [.33, .54, .07, .06]
+bottom = 1
 width = .2
-colors = [[.1, .3, .5], [.1, .3, .3], [.1, .3, .7], [.1, .3, .9]]
 
-for j in range(len(ratios)):
-    height = ratios[j]
-    ax2.bar(xpos, height, width, bottom=bottom, color=colors[j])
-    ypos = bottom + ax2.patches[j].get_height() / 2
-    bottom += height
-    ax2.text(xpos, ypos, "%d%%" % (ax2.patches[j].get_height() * 100),
-             ha='center')
+# Adding from the top matches the legend without have to explicitly re-order
+# the legend.
+for j, height in enumerate(reversed(approved_ratios)):
+    bottom -= height
+    bc = ax2.bar(0, height, width, bottom=bottom, color='C0',
+                 alpha=0.1 + 0.25 * j)
+    ax2.bar_label(bc, labels=["%d%%" % (height * 100,)], label_type='center')
```

Comment:
This breaks the association of `approved_ratios` to the legend labels.

Anyway, `legend(labels)` is a bad idea because you rely on the insertion order of artists.
I suggest you do

```
age_ratios = [.33, .54, .07, .06]
age_labels = ['Under 35', '35-49', '50-65', 'Over 65']
for j, height, label in ...:
    ....
    ax2.bar(..., label=label)

ax2.legend()
```

Reply:
I'm not sure what you mean about breaking the association, but I applied the change anyway.